interfaces/builtin/unity8.go

```diff
@@ -0,0 +1,161 @@
+// -*- Mode: Go; indent-tabs-mode: t -*-
+
+/*
+ * Copyright (C) 2016-2017 Canonical Ltd
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License version 3 as
+ * published by the Free Software Foundation.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ *
+ */
+
+package builtin
+
+import (
+	"bytes"
+	"fmt"
+
+	"github.com/snapcore/snapd/interfaces"
+	"github.com/snapcore/snapd/interfaces/dbus"
+)
+
+var unity8ConnectedPlugAppArmor = []byte(`
+# Description: Can access unity8 desktop services
+
+#include <abstractions/dbus-session-strict>
+
+# Fonts
+#include <abstractions/fonts>
+/var/cache/fontconfig/   r,
+/var/cache/fontconfig/** mr,
+
+# The snapcraft desktop part may look for schema files in various locations, so
+# allow reading system installed schemas.
+/usr/share/glib*/schemas/{,*}              r,
+/usr/share/gnome/glib*/schemas/{,*}        r,
+/usr/share/ubuntu/glib*/schemas/{,*}       r,
+
+# URL dispatcher. All apps can call this since:
+# a) the dispatched application is launched out of process and not
+#    controllable except via the specified URL
+# b) the list of url types is strictly controlled
+# c) the dispatched application will launch in the foreground over the
+#    confined app
+dbus (send)
+     bus=session
+     path=/com/canonical/URLDispatcher
+     interface=com.canonical.URLDispatcher
+     member=DispatchURL
+     peer=(name=com.canonical.URLDispatcher,label=###SLOT_SECURITY_TAGS###),
+
+# This is needed when the app is already running and needs to be passed in
+# a URL to open. This is most often used with content-hub providers and
+# url-dispatcher, but is actually supported by Qt generally.
+dbus (receive)
+     bus=session
+     path=/###PLUG_DBUS_APPIDS###
+     interface=org.freedesktop.Application
+     member=Open
+     peer=(label=###SLOT_SECURITY_TAGS###),
```

Comment:
I'd prefer that we have the intended rule and the documented workaround rule for transparency. Ie:
```
dbus (receive)
     bus=session
     path=/@{PROFILE_DBUS}
     interface=org.freedesktop.Application
     member=Open
     peer=(label=###SLOT_SECURITY_TAGS###),

# FIXME: workaround rule while UAL is using click-style AppID. This does not
# provide isolation since 'foo' can access 'barfoobaz'
dbus (receive)
     bus=session
     path=/###PLUG_DBUS_APPIDS###
     interface=org.freedesktop.Application
     member=Open
     peer=(label=###SLOT_SECURITY_TAGS###),
```